lib/processor/redis.go

```diff
@@ -0,0 +1,290 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package processor
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/Jeffail/benthos/v3/lib/log"
+	"github.com/Jeffail/benthos/v3/lib/message"
+	"github.com/Jeffail/benthos/v3/lib/metrics"
+	"github.com/Jeffail/benthos/v3/lib/types"
+	"github.com/Jeffail/benthos/v3/lib/util/text"
+	"github.com/go-redis/redis"
+	"github.com/opentracing/opentracing-go"
+)
+
+//------------------------------------------------------------------------------
+
+func init() {
+	Constructors[TypeRedis] = TypeSpec{
+		constructor: NewRedis,
+		description: `
+TODO: docs
+		`,
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// RedisConfig contains configuration fields for the Redis processor.
+type RedisConfig struct {
+	URL         string `json:"url" yaml:"url"`
+	Parts       []int  `json:"parts" yaml:"parts"`
+	Operator    string `json:"operator" yaml:"operator"`
+	Key         string `json:"key" yaml:"key"`
+	Value       string `json:"value" yaml:"value"`
+	Prefix      string `json:"prefix" yaml:"prefix"`
+	Expiration  string `json:"expiration" yaml:"expiration"`
+	Retries     int    `json:"retries" yaml:"retries"`
+	RetryPeriod string `json:"retry_period" yaml:"retry_period"`
+}
+
+// NewRedisConfig returns a RedisConfig with default values.
+func NewRedisConfig() RedisConfig {
+	return RedisConfig{
+		URL:         "tcp://localhost:6379",
+		Prefix:      "",
+		Expiration:  "24h",
+		Retries:     3,
+		RetryPeriod: "500ms",
+		Parts:       []int{},
+		Operator:    "sadd",
+		Key:         "",
+		Value:       "",
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// Redis is a processor that performs redis operations
+type Redis struct {
+	parts []int
+	conf  Config
+	log   log.Modular
+	stats metrics.Type
+
+	key               *text.InterpolatedString
+	valueBytes        []byte
```

Comment:
You can do the same for `valueBytes` as you do for `key` and use https://godoc.org/github.com/Jeffail/benthos/lib/util/text#NewInterpolatedBytes